dist/snapcraft.yaml

```
@@ -0,0 +1,157 @@
+##########
+# Freeciv21 - snapcraft.yaml
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2023 James Robertson <jwrober@gmail.com>
+#
+# Build Steps:
+#   - (once) sudo snap install snapcraft --classic
+#   - (once) sudo snap install multipass
+#   - mkdir -p <source_dir>/build/snap
+#   - cp <source_dir>/dist/snapcraft.yaml <source_dir>/build/snap
+#   - cp <source_dir>/data/icons/128x128/freeciv-client.png <source_dir>/build/snap
+#   - cd <source_dir>/build
+#   - snapcraft
+#   - sudo snap install --devmode ./freeciv21_*.snap
+#
+##########
+
+name: freeciv21
+summary: Freeciv21 - Freeciv for the 21st Century
+description: |
+  Freeciv21 is a free and open source empire-building strategy game inspired by the history of human civilization. The game commences in prehistory and your mission is to lead your tribe from the Stone Age to the Space Age.
+  This snap package contains everything you need to play local single-player games against the computer as well as connect to online multi-player games against human opponents.
+  Freeciv21 is based on the Qt framework and brought to you by the team at Longturn.net.
+
+# TODO: At some point, will want to integrate into cmake process and configure_file()
+version: '3.0-rc.1'
+grade: stable
+
+contact: longturn.net@gmail.com
+issues: https://github.com/longturn/freeciv21/issues/new/choose
+confinement: strict
+compression: lzo
+
+# The base snap is the execution environment for this snap and is based on Ubuntu 20.04
+#  core22 (Ubuntu 22.04) is available, but not all Qt features are ready yet
+base: core20
+
+# Assummes 'snapcraft' is called from a directory above the snap directory
+#  and the file was copied from <source_dir>/data/icons/128x128
+icon: snap/freeciv21-client.png
+
+parts:
+  freeciv21:
+    plugin: cmake
+    source: https://github.com/longturn/freeciv21.git
+    source-type: git
+    # TODO: integrate into cmake process and configure_file()
+    source-branch: stable
+    cmake-generator: Ninja
+    cmake-parameters:
+      - "-DCMAKE_INSTALL_PREFIX=/usr"
+      - "-DCMAKE_BUILD_TYPE=Release"
+    # packages/libraries needed at build time
+    build-snaps:
+      - kde-frameworks-5-qt-5-15-core20-sdk
+      - kde-frameworks-5-qt-5-15-core20
+    build-packages:
+      - git
+      - cmake
+      - ninja-build
+      - g++
+      - python3
+      - gettext
+      - qtbase5-dev
+      - libqt5svg5-dev
+      - libkf5archive-dev
+      - liblua5.3-dev
+      - libsqlite3-dev
+      - libsdl2-mixer-dev
+    # packages/libraries needed at run time
+    stage-snaps:
+      - kde-frameworks-5-qt-5-15-core20
+    stage-packages:
+      - fonts-linuxlibertine
+      - libdw1
+      - liblua5.3-0
+      - libsdl2-2.0-0
+      - libsdl2-mixer-2.0-0
+      - libsqlite3-0
+      - libunwind8
+      - python3
+      - libreadline8
+      - zlib1g
```

Some of those are not needed:
```suggestion
      - liblua5.3-0
      - libsdl2-2.0-0
      - libsdl2-mixer-2.0-0
```